sympy/sets/ordinals.py

```diff
@@ -0,0 +1,278 @@
+from sympy.core import Basic, Integer
+from sympy.core.compatibility import with_metaclass
+from sympy.core.singleton import Singleton, S
+import operator
+
+class OmegaPower(Basic):
+    """
+    Represents ordinal exponential and multiplication terms one of the
+    building blocks of the Ordinal class.
+    In OmegaPower(a, b) a represents exponent and b represents multiplicity.
+    """
+    def __new__(cls, a, b):
+        if isinstance(b, int):
+            b = Integer(b)
+        if not isinstance(b, Integer) or b <= 0:
+            raise TypeError("multiplicity must be a positive integer")
+
+        if not isinstance(a, Ordinal):
+            a = Ordinal.convert(a)
+
+        return Basic.__new__(cls, a, b)
+
+    @property
+    def exp(self):
+        return self.args[0]
+
+    @property
+    def mult(self):
+        return self.args[1]
+
+    def _compare_term(self, other, op):
+        if self.exp == other.exp:
+            return op(self.mult, other.mult)
+        else:
+            return op(self.exp, other.exp)
+
+    def __eq__(self, other):
+        if not isinstance(other, OmegaPower):
+            try:
+                other = OmegaPower(0, other)
+            except TypeError:
+                return NotImplemented
+        return self.args == other.args
+
+    __hash__ = Basic.__hash__
+
+    def __lt__(self, other):
+        if not isinstance(other, OmegaPower):
+            try:
+                other = OmegaPower(0, other)
+            except TypeError:
+                return NotImplemented
+        return self._compare_term(other, operator.lt)
+
+class Ordinal(Basic):
+    """
+    Represents ordinals in Cantor normal form.
+
+    Internally, this class is just a list of instances of OmegaPower
+    Examples
+    ========
+    >>> from sympy.sets import Ordinal, ord0, OmegaPower
+    >>> from sympy.sets.ordinals import omega
+    >>> w = omega
+    >>> w.is_limit_ordinal
+    True
+    >>> Ordinal(OmegaPower(w + 1 ,1), OmegaPower(3, 2)) #doctest: +SKIP
```

Comment:
Why should doctests be skipped?

Reply:
without skip , its failing.
```
Failed example:
    Ordinal(OmegaPower(w + 1 ,1), OmegaPower(3, 2))
Expected:
    w**(w + 1) + w**3*2
Got:
    Ordinal(OmegaPower(Ordinal(OmegaPower(Ordinal(OmegaPower(OrdinalZero(), 1)), 1), OmegaPower(OrdinalZero(), 1)), 1), OmegaPower(Ordinal(OmegaPower(OrdinalZero(), 3)), 2))
```

Reply:
It seems that Python doctest will pass if a blank line is added before "References".

    python -m doctest sympy/sets/ordinals.py

SymPy doctest is using a different printer, namely `sstrrepr`. It will probably also pass if the proper printer is added to `StrPrinter` ( str.py, in alphabetic order if possible):

    def _print_Ordinal(self, expr):
        return expr.__str__()